jetson/nav/search/searchStateMachine.cpp

```diff
@@ -10,217 +10,166 @@
+            }
+        }
     }
-    return NavState::SearchTurn;
-} // executeSearchDrive()
-
-// Executes the logic for turning to the target.
-// If the rover loses the target, will continue to turn using last known angles.
-// If the rover finishes turning to the target, it goes into waiting state to
-// give CV time to relocate the target
-// Else the rover continues to turn to to the target.
-NavState SearchStateMachine::executeTurnToTarget() {
-    std::shared_ptr<StateMachine> sm = mStateMachine.lock();
-    std::shared_ptr<Rover> rover = sm->getRover();
+    return NavState::Search;
+} // executeSearch()
 
-    if (rover->leftCacheTarget().distance == mRoverConfig["navThresholds"]["noTargetDist"].GetDouble()) {
-        return NavState::SearchTurn;
-    }
-    if (rover->turn(rover->leftCacheTarget().bearing + rover->odometry().bearing_deg)) {
-        return NavState::DriveToTarget;
-    }
-    updateTargetDetectionElements(rover->leftCacheTarget().bearing, rover->odometry().bearing_deg);
-    return NavState::TurnToTarget;
-} // executeTurnToTarget()
-
-// Executes the logic for driving to the target.
-// If the rover loses the target, it continues with search by going to
-// the last point before the rover turned to the target
-// If the rover detects an obstacle and is within the obstacle 
-// distance threshold, it proceeds to go around the obstacle.
-// If the rover finishes driving to the target, it moves on to the next Waypoint.
-// If the rover is on course, it keeps driving to the target.
-// Else, it turns back to face the target.
 NavState SearchStateMachine::executeDriveToTarget() {
     std::shared_ptr<StateMachine> sm = mStateMachine.lock();
     std::shared_ptr<Environment> env = sm->getEnv();
     std::shared_ptr<Rover> rover = sm->getRover();
 
-    // Definitely cannot find the target
-    if (rover->leftCacheTarget().distance == mRoverConfig["navThresholds"]["noTargetDist"].GetDouble()) {
-        std::cerr << "Lost the target\n";
-        return NavState::SearchTurn;
-    }
+    double dt = sm->getDtSeconds();
+    Waypoint lastWaypoint = sm->getCourseState()->getLastCompletedWaypoint();
+    Target const& leftTarget = env->getLeftTarget();
+    Target const& rightTarget = env->getRightTarget();
+    double currentBearing = rover->odometry().bearing_deg;
 
-    // Obstacle Detected
-    if (isObstacleDetected(rover, env)
-        && !isTargetReachable(rover, env, mRoverConfig)
-        && isObstacleInThreshold(rover, env, mRoverConfig)) {
-        Obstacle obstacle = env->getObstacle();
-        sm->updateObstacleElements(obstacle.bearing, obstacle.rightBearing, obstacle.distance);
-        return NavState::SearchTurnAroundObs;
+    double distance, bearing;
+    if (lastWaypoint.gate) {
+        if (env->hasGateLocation()) {
+            return NavState::BeginGateSearch;
+        }
+        if (leftTarget.id >= 0 && rightTarget.id >= 0) {
+            if (leftTarget.distance < rightTarget.distance) {
```

Comment:
This block from line 95 to about 100 is similar to how I had made it go to the closest post, but it resulted in the weird infinite loop between `TurnToTarget` and `DriveToTarget`. Did you test a case where we'd see a post that is supposed to go into the right side first (which would make it go into `leftTarget`) and then we'd see two posts, causing the one in the left to go to the right?